frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+import Dispatcher from '../dispatchers/GlossaryDispatcher';
+import assign from 'object-assign';
+import {EventEmitter} from 'events';
+import {Promise} from 'es6-promise';
+import Request from 'superagent';
+import Configs from '../constants/Configs';
+import {GlossaryActionTypes} from '../constants/ActionTypes';
+import GlossaryHelper from '../utils/GlossaryHelper'
+import StringUtils from '../utils/StringUtils'
+import DateHelpers from '../utils/DateHelper'
+import _ from 'lodash';
+
+var SIZE_PER_PAGE = 5000, CHANGE_EVENT = "change",
+    MAX_LISTENER = 100; //number of listener for GlossaryStore (default is 11)
+
+EventEmitter.prototype.setMaxListeners(MAX_LISTENER);
+
+var _state = {
+  canAddNewEntry: false,
+  canUpdateEntry: false,
+  localeOptions: [],
+  srcLocale: null,
+  selectedTransLocale: null,
+  locales: null,
+  glossary: {},
+  original_glossary: {},
+  glossaryResId: [],
+  page:1,
+  filter: '',
+  focusedRow: {
+    rowIndex: -1
+  },
+  uploadFile: {
+    show: false,
+    status : -1,
+    file: null,
+    transLocale: null
+  },
+  sort: {
+    src_content: true
+  },
+  totalCount: 0,
+  newEntry: {
+    term: '',
+    pos: '',
+    description: '',
+    isSaving: false,
+    show: false,
+    isAllowSave: false
+  }
+};
+
+function localesStatAPIUrl() {
+  return Configs.baseUrl + "/glossary/info" + Configs.urlPostfix
+}
+
+function loadLocalesStats() {
```

Comment:
This function could be extracted into a separate module (i.e. file) so that `GlossaryStore` is not so huge.